variables.tf

```diff
@@ -4,139 +4,269 @@ variable "name" {
+  type        = string
+  default     = "admin@default.com"  # Valor por defecto razonable
+}
+
+variable "sms_external_id" {
+  description = "SMS external ID"
+  type        = string
+  default     = ""  # Valor por defecto vacío si es opcional
+}
+
+variable "enable_sms_sns" {
+  description = "Boolean to enable/disable the creation of the SNS SMS policy"
+  type        = bool
+  default     = false
+}
+
+variable "sms_sns_caller_arn" {
+  description = "SNS caller ARN for Cognito SMS configuration"
+  type        = string
+  default     = null
+}
+
+variable "sms_sns_region" {
+  description = "SMS SNS region"
+  type        = string
+  default     = "us-east-1"  # Valor por defecto razonable
+}
+
+variable "email_recovery_name" {
+  description = "Email recovery mechanism name"
+  type        = string
+  default     = "verified_email"  # Valor por defecto razonable
+}
+
+variable "email_recovery_priority" {
+  description = "Email recovery mechanism priority"
+  type        = number
+  default     = 1  # Valor por defecto razonable
+}
+
+variable "phone_number_recovery_name" {
+  description = "Phone number recovery mechanism name"
+  type        = string
+  default     = "verified_phone_number"  # Valor por defecto razonable
+}
+
+variable "phone_number_recovery_priority" {
+  description = "Phone number recovery mechanism priority"
+  type        = number
+  default     = 2  # Valor por defecto razonable
+}
+
+variable "username_attributes" {
+  description = "Username attributes"
+  type        = list(string)
+  default     = ["email", "phone_number"]  # Valor por defecto razonable
+}
+
+variable "username_case_sensitive" {
+  description = "Username case sensitive"
+  type        = bool
+  default     = false  # Valor por defecto razonable
+}
+
+variable "password_minimum_length" {
+  description = "Password minimum length"
+  type        = number
+  default     = 8  # Valor por defecto razonable
+}
+
+variable "client_name" {
+  description = "Client name"
+  type        = string
+  default     = "default_client"  # Valor por defecto razonable
 }
 
 variable "callback_urls" {
-  type    = list(string)
-  default = [""]
+  description = "Callback URLs"
+  type        = list(string)
+  default     = ["http://localhost:3000"]  # Valor por defecto razonable
 }
 
 variable "logout_urls" {
-  type    = list(string)
-  default = [""]
+  description = "Logout URLs"
+  type        = list(string)
+  default     = ["http://localhost:3000"]  # Valor por defecto razonable
+}
+
+variable "region" {
+  description = "AWS region"
+  type        = string
+  default     = "us-east-1"  # Valor por defecto razonable
+}
+
+variable "domain" {
+  description = "Cognito domain"
+  type        = string
+  default     = "default.domain.com"  # Valor por defecto razonable
+}
+
+variable "certificate_arn" {
+  description = "Certificate ARN"
+  type        = string
+  default     = ""  # Valor por defecto vacío si es opcional
```

Comment:
Si se usa route53 y dominio custom, el certificado de ACM debe ser obligatorio

Reply:
The certificate_arn variable has been added and integrated into the aws_cognito_user_pool_domain resource configuration. This ensures that when using Route53 and a custom domain, a valid ACM certificate is specified, meeting the security and deployment requirements of the module.